src/main/scala/mesosphere/marathon/api/v2/validation/PodsValidation.scala

```diff
@@ -211,6 +212,9 @@ trait PodsValidation {
     }
     pod.secrets is empty or (valid(secretValidator) and featureEnabled(enabledFeatures, Features.SECRETS))
     pod.networks is valid(ramlNetworksValidator)
+    pod.networks is isTrue[Seq[Network]]("network name must be specified when container network is selected") { nets =>
```

Comment:
this logic is the same for both apps and pods. perhaps extract into a `NetworkValidation` class and `NetworkValidationMessages` class (used by both app and pod validation), and use the static message field (instead of a literal string) for the validation error